### ISSUE TITLE
 Incorrect coercion for alternatives of closed maps with optional keys

### ISSUE THREAD

**Issue by opqdonut** [metosin/malli#1112]
Originally reported as metosin/reitit#632. Moving it here because it's a malli bug.

Similar to metosin/reitit#407, except that the maps are closed and contain optional keys. For example,

```clj
(malli.core/decode
  [:or
   [:map
    {:closed true}
    [:x {:optional true} int?]]
   [:map
    {:closed true}
    [:y {:optional true} keyword?]]]
  {:y :foo}
  malli.transform/strip-extra-keys-transformer)
```

should return `{:y :foo}` but actually returns `{}`.

**Comment by opqdonut** [comment]
Previously, #201

**Comment by ikitommi** [comment]
Is it incorrect or just unfortunate? `strip-extra-keys-transformer` strips keys not defined in schema and returns the first valid branch. Would the correct behavior be "select a branch with least amount of changes to the original data"?

**Comment by opqdonut** [comment]
You're right, maybe the behaviour is correct. A `:multi` might be the right call here.
